From: SeongJae Park <sj@kernel.org>
Date: Sat, 13 Dec 2025 01:19:17 -0800
Subject: [PATCH] mm/damon/core: elaborate access reports dropping behavior

The access reports array is intentionally dropping all existing access
reports when the array is full.  It is an intended behavior for the
simplicity of the code that only exist for high level discussion and
early testing.  But the poor documentation is causing [1] some
confusions.  Add a comment elaborating the behavior.  The comment on the
function should explain the reason of the behavior well.

[1] https://lore.kernel.org/20251213055334.51806-1-sj@kernel.org

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 04c7e5f4c294..00a0b6a60de0 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1749,6 +1749,7 @@ void damon_report_access(struct damon_access_report *report)
 	if (!mutex_trylock(&damon_access_reports_lock))
 		return;
 	dst = &damon_access_reports[damon_access_reports_len++];
+	/* just drop all existing reports in favor of simplicity. */
 	if (damon_access_reports_len == DAMON_ACCESS_REPORTS_CAP)
 		damon_access_reports_len = 0;
 	*dst = *report;
-- 
2.47.3

